src/main/java/scripts/versionscripts/ReadFromTerraHelmfileRepo.java

```diff
@@ -66,38 +64,11 @@ public VersionScriptResult determineVersion(ServerSpecification server) throws E
     String wsmHelmChartVersion =
         helmRelease.getReleases().get("workspacemanager").getChartVersion().orElse("");
 
-    return new VersionScriptResult.Builder()
-        .wsmHelmAppVersion(wsmHelmAppVersion)
-        .wsmHelmChartVersion(wsmHelmChartVersion)
-        .build();
-  }
-
-  /**
-   * @param from a Java Map instance that represents the source of updates
-   * @param to a Java Map instance to receive updates from source map
-   * @return the same 'to' instance with merged keys from the source map
-   */
-  private Map<String, HelmRelease.HelmReleaseVersion> merge(
-      Map<String, HelmRelease.HelmReleaseVersion> from,
-      Map<String, HelmRelease.HelmReleaseVersion> to) {
-    from.forEach(
-        (app, version) ->
-            to.merge(
-                app,
-                version,
-                (toVersion, fromVersion) ->
-                    new HelmRelease.HelmReleaseVersion(
-                        fromVersion.getEnabled() != null && fromVersion.getEnabled().isPresent()
-                            ? fromVersion.getEnabled()
-                            : toVersion.getEnabled(),
-                        fromVersion.getChartVersion() != null
-                                && fromVersion.getChartVersion().isPresent()
-                            ? fromVersion.getChartVersion()
-                            : toVersion.getChartVersion(),
-                        fromVersion.getAppVersion() != null
-                                && fromVersion.getAppVersion().isPresent()
-                            ? fromVersion.getAppVersion()
-                            : toVersion.getAppVersion())));
-    return to;
+    return Arrays.asList(
+        new VersionScriptResult.HelmVersion.Builder()
+            .appName("workspacemanager")
```

Comment:
Since this is just for WSM, perhaps you can take the app name as a parameter to this script? That way we could reuse this for other apps also, since I expect the helm merge logic would be identical.

Reply:
Replaced hardcoded value by `appName` property.